hist/hist/src/TFormula.cxx

```diff
@@ -3236,6 +3242,107 @@ void TFormula::GradientPar(const Double_t *x, Double_t *result)
    }
 }
 
+/// returns true on success.
+bool TFormula::GenerateHessianPar()
+{
+   // We already have generated the hessian.
+   if (fHessMethod)
+      return true;
+
+   if (!HasHessianGenerationFailed()) {
```

Comment:
Could this be converted into an early return, in place of a large, indented block?